PowerPointLabs/Test/FunctionalTest/ShapesLabTest.cs

```diff
@@ -87,5 +88,27 @@ private void DoubleClick(Control target)
             Point pt = target.PointToScreen(new Point(target.Width/2, target.Height/2));
             MouseUtil.SendMouseDoubleClick(pt.X, pt.Y);
         }
+
+        private void TestSaveShapesToShapesLabWithAddShapesButton(IShapesLabController shapesLab)
+        {
+            PpOperations.SelectSlide(6);
+            PpOperations.SelectShapesByPrefix("selectMeNow");
+
+            shapesLab.ClickAddShapeButton();
+
+            Microsoft.Office.Interop.PowerPoint.Slide actualSlide = PpOperations.SelectSlide(7);
+            IShapesLabLabeledThumbnail addedThumbnail = shapesLab.GetLabeledThumbnail("selectMeNow1");
+            addedThumbnail.FinishNameEdit();
+            // add shapes back
+            DoubleClick(addedThumbnail as Control);
+            Microsoft.Office.Interop.PowerPoint.ShapeRange shapes = PpOperations.SelectShapesByPrefix("Group selectMeNow1");
+            Assert.IsTrue(shapes.Count > 0, "Failed to add shapes from Shapes Lab." +
```

Comment:
Is there any reason for the flakiness? How often does the FT fail?

Reply:
Actually I copied the comment from the above function `TestSaveShapesToShapesLab`, but to be honest it hasn't failed for me before. I'll refactor this.